### ISSUE TITLE
Roll/Unroll parent page bug in the treeview 

### ISSUE THREAD

**Issue by tehkonst** [zim-desktop-wiki/zim-desktop-wiki#1820]
**Describe the bug**
When I click a small triangle near a parent page I can't unroll it back

![unroll_bug](https://user-images.githubusercontent.com/2618162/146309166-035250db-62f9-40f8-b399-49cb910a945b.gif)

1. Choose any note on the second level
2. Click the triangle near the parent entry
3. Then click again

And another one:
![unroll_bug2](https://user-images.githubusercontent.com/2618162/146310961-1ffd180f-14bd-4162-8685-61fc46cb0a56.gif)
1. Unroll any non-current branch
2. The click on it and make it active
3. Then click the triangle several times

**Expected behavior**
The treeview elements have to be rolled up and unrolled as expected.

**Environment (please complete the following information):**
 - Version 0.74.3 -- ppa
 - xUbuntu 20.04

**Comment by introt** [comment]
Confirmed - after opening a page and closing its branch, it can no longer be opened. Nice catch!

**Comment by introt** [comment]
Keybindings don't stop working, and after expanding with one, the branch becomes collapsible once more- yet can't be expanded again using the button.

**Comment by nelas** [comment]
I also ran into this issue when `Edit > Preferences > Plugins > Page Index > Configure` is like that:

```
[x] Automatically expand sections on open page
[ ] Automatically collapse sections on close page
```

If the automatic collapse option is also activated, clicking on the triangle of the current page neither expands nor collapses the tree (it stays expanded).

**Comment by rscmbbng** [comment]
Can confirm issue as well!

`$ zim --version
zim 0.74.2
`

**Comment by mke21** [comment]
Zim version 0.74.3 on windows has this as well. Arrow keys still work though.

**Comment by prahal** [comment]
0.74.3 on Debian amd64. Reproducible with both gtk3 3.24.24-4+deb11u2 and 3.24.34-1.
Note that here this is only reproducible on a treeview item that is selected (current page in the index).
Switching to a subpage or a sibling let me fold/unfold it anew.

**Comment by sojusnik** [comment]
After [this](https://github.com/zim-desktop-wiki/zim-desktop-wiki/pull/2113) merge the roll/unroll issue is fixed, but I also observe that when jumping to another page through the history bar or bookmarks bar or the "jump to" window etc., meaning NOT using the index/tag side pane to jump to another page, then the current page with its subpages isn't automatically folded, when switching to another page with subpages. Autofolding/-unfolding works reliably only when using the index/tag side pane. Of course, `Automatically expand sections on open page` and `Automatically collapse sections on close page` have to be enabled in the options of the index/tag side pane.

Is this by design?

I would prefer if the folding/unfolding mechanism would work consistently, regardless of whether the side pane or other methods are used to jump to pages.

BTW, manually folded/unfolded pages shouldn't be affected by the automatic folding/unfolding mechanism switchable in the settings (as mentioned above), as it is already done now.

**Comment by introt** [comment]
Related issue: #2119

**Comment by prahal** [comment]
@sojusnik you should open a new issue for this inconsistent behavior. This is another code path than the current issue which is fixed.

> After [this](https://github.com/zim-desktop-wiki/zim-desktop-wiki/pull/2113) merge the roll/unroll issue is fixed, but I also observe that when jumping to another page through the history bar or bookmarks bar or the "jump to" window etc., meaning NOT using the index/tag side pane to jump to another page, then the current page with its subpages isn't automatically folded, when switching to another page with subpages. Autofolding/-unfolding works reliably only when using the index/tag side pane. Of course, `Automatically expand sections on open page` and `Automatically collapse sections on close page` have to be enabled in the options of the index/tag side pane.
> 
> Is this by design?
> 
> I would prefer if the folding/unfolding mechanism would work consistently, regardless of whether the side pane or other methods are used to jump to pages.
> 
> BTW, manually folded/unfolded pages shouldn't be affected by the automatic folding/unfolding mechanism switchable in the settings (as mentioned above), as it is already done now.

**Comment by sojusnik** [comment]
> @sojusnik you should open a new issue for this inconsistent behavior. This is another code path than the current issue which is fixed.

[Done](https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/2205)

**Comment by prahal** [comment]
This issue is fixed released in 0.75.1.  Should be closed it seems.
